src/MaestroConfiguration/src/Microsoft.DotNet.MaestroConfiguration.Client/ConfigurationRepositoryManager.cs

```
@@ -86,133 +114,170 @@ await CreatePullRequest(
             configurationRepo,
             parameters.RepositoryUri,
             workingBranch,
-            newSubscriptionFilePath);
+            filePath);
 
-        // If we have a branch that hasn't been ingested yet, we need to check for equivalent subscriptions in the file
-        var equivalentInFile = subscriptionsInFile.FirstOrDefault(s => s.IsEquivalentTo(subscription));
+        var equivalentInFile = yamlModelsInFile.FirstOrDefault(m => isEquivalent(m, yamlModel));
         if (equivalentInFile != null)
         {
-            throw new ArgumentException($"Subscription {equivalentInFile.Id} with equivalent parameters already exists in '{newSubscriptionFilePath}'.");
+            throw new ArgumentException(getDuplicateError(equivalentInFile, filePath));
         }
 
-        subscriptionsInFile.Add(subscription);
+        yamlModelsInFile.Add(yamlModel);
         await CommitConfigurationDataAsync(
             configurationRepo,
             parameters.RepositoryUri,
             workingBranch,
-            newSubscriptionFilePath,
-            subscriptionsInFile,
-            new SubscriptionYamlComparer(),
-            $"Add new subscription ({subscription.Channel}) {subscription.SourceRepository} => {subscription.TargetRepository} ({subscription.TargetBranch})"); ;
+            filePath,
+            yamlModelsInFile,
+            comparer,
+            commitMessage);
     }
 
-    private async Task DeleteSubscriptionInternalAsync(
+    /// <summary>
+    /// Generic method to delete a configuration object from the configuration repository.
+    /// </summary>
+    private async Task DeleteModelInternalAsync<T, TKey>(
         ConfigurationRepositoryOperationParameters parameters,
         IGitRepo configurationRepo,
         string workingBranch,
-        SubscriptionYaml subscription)
+        T yamlModel,
+        Func<T, TKey> getKey,
+        Func<T, string, string, string, string> getNotFoundError,
```

Same - we could throw a specific exception instead